src/DbConnection.cpp

```diff
@@ -66,26 +65,26 @@ void DbConnection::connect(const Nullable<std::string>& host, const Nullable<std
   LOG_VERBOSE;
 
   if (!mysql_real_connect(this->pConn_,
-                          host.isNull() ? NULL : as<std::string>(host).c_str(),
-                          user.isNull() ? NULL : as<std::string>(user).c_str(),
-                          password.isNull() ? NULL : as<std::string>(password).c_str(),
-                          db.isNull() ? NULL : as<std::string>(db).c_str(),
+                          Rf_isNull(host) ? NULL : cpp11::as_cpp<std::string>(host).c_str(),
+                          Rf_isNull(user) ? NULL : cpp11::as_cpp<std::string>(user).c_str(),
+                          Rf_isNull(password) ? NULL : cpp11::as_cpp<std::string>(password).c_str(),
+                          Rf_isNull(db) ? NULL : cpp11::as_cpp<std::string>(db).c_str(),
                           port,
-                          unix_socket.isNull() ? NULL : as<std::string>(unix_socket).c_str(),
+                          Rf_isNull(unix_socket) ? NULL : cpp11::as_cpp<std::string>(unix_socket).c_str(),
                           client_flag)) {
     std::string error = mysql_error(this->pConn_);
     mysql_close(this->pConn_);
     this->pConn_ = NULL;
 
-    stop("Failed to connect: %s", error.c_str());
+    cpp11::stop("Failed to connect: %s", error.c_str());
   }
 }
 
 void DbConnection::disconnect() {
   if (!is_valid()) return;
 
   if (has_query()) {
-    warning(
+    cpp11::warning(
```

Comment:
I think in case of removing warning from destructor, it's also would be good to remove warning from `DbConnection::disconnect` method, since it's calling in destructor. There just no such test case as I think.